plutus-core/plutus-core/src/PlutusCore/Pretty/Extra.hs

```diff
@@ -1,19 +1,52 @@
+{-# LANGUAGE ConstraintKinds       #-}
 {-# LANGUAGE FlexibleInstances     #-}
+{-# LANGUAGE InstanceSigs          #-}
 {-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE TypeApplications      #-}
 {-# LANGUAGE UndecidableInstances  #-}
 
 {-# OPTIONS_GHC -fno-warn-orphans #-}
 
-module PlutusCore.Pretty.Extra () where
+module PlutusCore.Pretty.Extra
+    ( PrettyParens
+    , prettyParens
+    ) where
 
 import PlutusPrelude
 
+import Control.Monad.Trans.Reader
 import Data.Map (Map)
 import Data.Map qualified as Map
+import Data.Profunctor
 import Data.Set (Set)
 import Data.Set qualified as Set
+import Text.PrettyBy.Fixity
 import Text.PrettyBy.Internal
 
+instance Profunctor InContextM where
+    lmap
+        :: forall config' config a.
+           (config' -> config)
+        -> InContextM config a
+        -> InContextM config' a
+    lmap = coerce (withReader :: (config' -> config) -> Reader config a -> Reader config' a)
+    {-# INLINE lmap #-}
+
+    rmap
+        :: forall config a b.
+           (a -> b)
+        -> InContextM config a
+        -> InContextM config b
+    rmap = coerce (mapReader :: (a -> b) -> Reader config a -> Reader config b)
+    {-# INLINE rmap #-}
```

Comment:
Is this instance used anywhere?

Reply:
[Yes](https://github.com/input-output-hk/plutus/blob/baa58bc6a21d840d44737f3850e406e247d173f9/plutus-core/plutus-core/src/PlutusCore/Core/Instance/Pretty/Readable.hs#L61).